Current status

Cabel496 opened the pull request:
Step 1: Making the code compile
- I set `const bool hot_standby_ = false`
- The constructor of `DiskLogConsumerTask` is expecting a file name (`log_file_path`), I am not sure what that is so I set it to empty for `log_manager.cpp`

Step 2: Separating Postgres classes from ITP
- I deleted `abstract_command_factory` as I don't see anything being shared between ITP and Postgres.
- Now `ITPNetworkCommand` and `PostgresNetworkCommand` both inherit from `AbstractNetworkCommand`. In this way, I will be able to separate out the `PostgresPacketWriter`.
- Update: Since we still need `AbstractNetworkCommand', I have added this back. In order for this to work, I have updated network command classes to return 'AbstractNetworkCommand' pointers. I also added an `AbstractPacketWriter` for the sake of sharing more common functions and allowing the `transition::Exec` function to look the same across ITP/Postgres commands.

jrolli committed:
Make Jenkins print hostname, increase build parallelism on Jenkins Linux machines. (#525)

Commit:
Insert contention problems (#516)

tli2 committed:
GC queue and Deferred Action queue Ordering (#527)

mbutrovich committed:
Update lookback. (#542)

mbutrovich committed:
Change catalog's #define'd constants to be constexpr instead (#538)

amlatyrngom committed:
Tpl without codegen (#398)

This is an amazing PR that brings in the TPL execution engine without codege. There are way too many things to mention here. This is a massive commit.

Special thanks to @amlatyrngom + @pmenon for making this happen! #398

mbutrovich committed:
Basic catalog microbenchmarks (#545)

mbutrovich committed:
Catalog: reuse ProjectedRowInitializers and maps (#546)

Cabel496 committed:
Fix depency in network classes and functions for storage logger classes

Cabel496 committed:
Add helper function to extract referenced column oids from AbstractExpression (#539)

GustavoAngulo committed:
Better GetIndexes method (#553)

tanujnay112 committed:
Added Index OID vector to InsertPlanNode (#543)

GustavoAngulo committed:
Recovery Manager (#459)

mbutrovich committed:
Hash index based on libcuckoo (#485)

Cabel496 committed:
updated test

mbutrovich committed:
Update LLVM_DIR path for llvm@8 Homebrew package (#559)

mbutrovich committed:
Move coverage to Jenkins from Travis (#551)

GustavoAngulo committed:
Add recovery benchmark to nightly runs and new index recovery benchmark (#558)

Commit:
use nproc instead of hardcoded number for number of make jobs (#562)

Cabel496 committed:
Seprate ITP from postgres

Cabel496 committed:
use ITP classes for network commands

mbutrovich committed:
Multiplatform thread CPU timer (#563)

apavlo committed:
Add missing documentation for typedefs (#570)

* Added missing documentation for two typedefs that Doxygen on Travis occassionally complains about #569

* Updated .gitignore to hide libpqxx files in root dir

mbutrovich committed:
Simplified ThreadCPUTimer test. (#573)

GustavoAngulo commented on src/include/network/abstract_command_factory.h:
```diff
@@ -1,24 +0,0 @@
-   * @param packet the input packet
-   * @return a shared_ptr to the converted command
-   */
-  virtual std::shared_ptr<AbstractNetworkCommand> PacketToCommand(InputPacket *packet) = 0;
```
From the PR description:

> I deleted abstract_command_factory as I don't see anything being shared between ITP and Postgres.

Both `PostgresCommandFactory` and `ITPCommandFactory` can share the `PacketToCommand` method. Since `PostgresNetworkCommand` and `ITPNetworkCommand` both inherit from `AbstractNetworkCommand`, both of the derived methods can have the same return type

Cabel496 replied:
I see; I have added this class back.

GustavoAngulo commented on src/include/network/itp/itp_command_factory.h:
```diff
@@ -1,21 +1,22 @@
-#include "network/abstract_command_factory.h"
+#include "network/itp/itp_network_commands.h"
+
+#define MAKE_COMMAND(type) std::static_pointer_cast<PostgresNetworkCommand, type>(std::make_shared<type>(packet))
```
nit: Should be `MAKE_ITP_COMMAND`.

GustavoAngulo reviewed (changes requested):
Overall looks good. Just one change related to the command factory. We want as much in the abstract class as possible to prevent future people from having vastly different APIs for these protocols.

apavlo committed:
AstDump Unit Test (#572)

Cabel496 committed:
Add changse to ITP

Cabel496 committed:
Add back factory

Cabel496 committed:
Switch to use AbstractNetworkCommands and add PacketWriter classes

Cabel496 committed:
Remove dead code

Cabel496 committed:
revert files under storage/writer_ahead_log

Cabel496 committed:
add debug message

Cabel496 committed:
add variable name

Cabel496 committed:
comment fixes

Cabel496 committed:
update PostgresPacketWriter

apavlo committed:
Cleaning up nightly microbench script (#565)

Cabel496 committed:
cleanup

Cabel496 committed:
fix conflict

Cabel496 committed:
Postgres message type refactor

Cabel496 committed:
include tests

Cabel496 committed:
fix build and add comments after merge

Cabel496 committed:
remove itp

Cabel496 committed:
format

mbutrovich committed:
Add `make clean` as final step of Jenkins builds to reduce the archived size builds. (#579)

Cabel496 committed:
clang

Cabel496 committed:
override

Cabel496 committed:
add back default

mbutrovich committed:
Metrics Manager (#506)

lmwnshn committed:
Parser/Plan Nodes/Catalog ManagedPointer and JSON refactor (#547)

Wan's magical rewrite of the parser to use the ManagedPointer.

This releasing Wan from his personal hell. This PR has been a serious detriment to his health and well being (and his love life!).

mbutrovich committed:
Modify TPC-C microbenchmark framework to use Catalog (#578)

mbutrovich committed:
ASAN flag wasn't being passed to the IR compiler (#550)

Cabel496 committed:
more refactor

Cabel496 committed:
remove hot_standby

mbutrovich committed:
Fix hashing DropTriggerPlanNode and UpdatePlanNode

Cabel496 committed:
build fix

Cabel496 committed:
comments

Cabel496 committed:
wut

mbutrovich committed:
Remove murmur3 from the codebase, bump xxHash version (#590)

Cabel496 committed:
Merge branch 'master' into internal_terrier_protocol_tianlei

Cabel496 committed:
set virtual

Cabel496 committed:
confusion

GustavoAngulo committed:
Increase recovery benchmark iterations (#591)

mbutrovich committed:
Merge branch 'master' into internal_terrier_protocol_tianlei

Cabel496 committed:
build fix and comments

Cabel496 committed:
build fix

Cabel496 committed:
is packet empty

Cabel496 committed:
use appendraw

Cabel496 committed:
format

Cabel496 committed:
test fix

Cabel496 committed:
change function

Cabel496 committed:
these asserts should use this function as well

Cabel496 committed:
Refactoring for network protocols (#574)

Separates ITP classes from Postgres and establishes abstract classes that will define future APIs

Cabel496 committed:
Merge remote-tracking branch 'upstream/master' into internal_terrier_protocol_tianlei
